first_run.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+# function to enable communication protocols such as SPI and I2C.
+enable_protocol() {
+    local protocol=$1
+    local get_cmd="get_${protocol}"
+    local do_cmd="do_${protocol}"
+
+    # Check the current status of the protocol
+    status=$(sudo raspi-config nonint "$get_cmd")
+    if [[ $? -ne 0 ]]; then
+        echo "Failed to get ${protocol} status. Ensure that raspi-config is installed."
+        exit 1
+    fi
```

Comment:
Check that you're on a rapsberry pi first.

Also I notice you're using bin/bash. I think the syntax is correct, I'm expecting all the checks to be in `[[ ]]` but just run this file on a random raspberry pi after you turn off i2c to ensure it works.

Reply:
Added check for rpi, tested on my pi and it works as expected!